subprojects/docs/src/docs/userguide/authoring-builds/introduction/writing_tasks.adoc

```diff
@@ -0,0 +1,241 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[tutorial_using_tasks]]
+= Tasks Basics
+
+The work that Gradle can do on a project is defined by one or more _tasks_.
+
+A task represents some atomic piece of work which a build performs.
+This might be compiling some classes, creating a JAR, generating Javadoc, or publishing some archives to a repository.
+
+You run a Gradle build using the `gradle` command or by invoking the Gradle Wrapper (`./gradlew' or `gradle.bat`) in your project directory.
+
+As part of the build process, Gradle finds all the tasks in your project coded in plugins or build scripts.
+
+[[sec:hello_world]]
+== Hello world
+
+Let's take a look at a simple "Hello World" task in a build script:
+
+====
+include::sample[dir="snippets/tutorial/hello/kotlin", files="build.gradle.kts[]"]
+include::sample[dir="snippets/tutorial/hello/groovy", files="build.gradle[]"]
+====
+
+You can execute the build script with `./gradlew -q hello`:
+
+====
+----
+$ ./gradlew -q hello
+include::{snippetsPath}/tutorial/hello/tests/hello.out[]
+----
+====
+
+[TIP]
+====
+The `-q` option suppresses Gradle's log messages, so that only the output of the tasks is shown.
+====
+
+In the example, the build script defines a single task, called `hello`, and adds an action to it.
+
+When you run `./gradlew hello`, Gradle executes the `hello` task, which in turn executes the action you've provided.
+The action is simply a block containing some code to execute: `println("Hello world!")`.
+
+== Available tasks
+
+Even if your build script is empty, Gradle provides a number of default tasks for a project which are listed by running `./gradlew tasks`:
+
+[source]
+----
+------------------------------------------------------------
+Tasks runnable from root project 'myTutorial'
+------------------------------------------------------------
+
+Build Setup tasks
+-----------------
+init - Initializes a new Gradle build.
+wrapper - Generates Gradle wrapper files.
+
+Help tasks
+----------
+buildEnvironment - Displays all buildscript dependencies declared in root project 'myTutorial'.
+...
+----
+
+Once we apply a plugin to our project such as the `application` plugin, the default tasks available include:
+
+[source]
+----
+Application tasks
+-----------------
+run - Runs this project as a JVM application
+
+Build tasks
+-----------
+assemble - Assembles the outputs of this project.
+build - Assembles and tests this project.
+
+...
+
+Documentation tasks
+-------------------
+javadoc - Generates Javadoc API documentation for the main source code.
+
+...
+
+Other tasks
+-----------
+compileJava - Compiles main Java source.
+
+...
+----
+
+This is because the `application` plugin code contains additional tasks which are now available in your project.
+
+== Task types
+
+There are two types of tasks:
+
+1. *Actionable tasks* have some action(s) attached to do work in your build: `compileJava`.
```

Comment:
💭 "Actionable" sounds like "able to be executed.  But lifecycle tasks can be executed too, they just do no work.  Is there a better word for these?